modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -725,8 +725,11 @@ private CompletableFuture<?> updateAssignmentInternal(ConfigurationNotificationE
 
                 placementDriver.updateAssignment(replicaGrpId, newConfiguration.peers().stream().map(Peer::consistentId).collect(toList()));
 
-                PendingComparableValuesTracker<HybridTimestamp> safeTime = new PendingComparableValuesTracker<>(new HybridTimestamp(1, 0));
-                PendingComparableValuesTracker<Long> storageIndexTracker = new PendingComparableValuesTracker<>(0L);
+                var safeTimeTracker = new PendingComparableValuesTracker<>(new HybridTimestamp(1, 0));
+                var storageIndexTracker = new PendingComparableValuesTracker<>(0L);
+
+                ((InternalTableImpl) internalTbl).updatePartitionSafeTimeTracker(partId, safeTimeTracker);
+                ((InternalTableImpl) internalTbl).updatePartitionStorageIndexTracker(partId, storageIndexTracker);
```

Comment:
These 2 methods are always called together. Does it make sense to merge them in one?

Reply:
Why not.